cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/executor/ExecutorTest.java

```diff
@@ -367,22 +367,55 @@ private static boolean verifyFutureError(Future<?> future, Class<? extends Throw
   }
 
   @Test
-  public void testExecutionKnobs() {
+  public void testSetExecutionProgressCheckIntervalMs() {
     KafkaCruiseControlConfig config = new KafkaCruiseControlConfig(getExecutorProperties());
-    assertThrows(IllegalStateException.class,
-                 () -> new Executor(config, null, new MetricRegistry(), EasyMock.mock(MetadataClient.class), null, null));
     Executor executor = new Executor(config, null, new MetricRegistry(), EasyMock.mock(MetadataClient.class),
                                      null, EasyMock.mock(AnomalyDetectorManager.class));
-
-    // Verify correctness of set/get requested execution progress check interval.
     long defaultExecutionProgressCheckIntervalMs = config.getLong(ExecutorConfig.EXECUTION_PROGRESS_CHECK_INTERVAL_MS_CONFIG);
-    assertEquals(defaultExecutionProgressCheckIntervalMs, executor.executionProgressCheckIntervalMs());
     long minExecutionProgressCheckIntervalMs = config.getLong(ExecutorConfig.MIN_EXECUTION_PROGRESS_CHECK_INTERVAL_MS_CONFIG);
+
+    // RequestedExecutionProgressCheckIntervalMs has to be larger than minExecutionProgressCheckIntervalMs
     executor.setRequestedExecutionProgressCheckIntervalMs(minExecutionProgressCheckIntervalMs);
     assertEquals(minExecutionProgressCheckIntervalMs, executor.executionProgressCheckIntervalMs());
     assertThrows(IllegalArgumentException.class,
                  () -> executor.setRequestedExecutionProgressCheckIntervalMs(minExecutionProgressCheckIntervalMs - 1));
 
+    // With requested ≈, the value should be capped at requestedExecutionProgressCheckIntervalMs, and
+    //  set to at least minExecutionProgressCheckIntervalMs
+    long requestedExecutionProgressCheckIntervalMs = 2 * defaultExecutionProgressCheckIntervalMs;
```

Comment:
if underlying the executor is not too slow to spin up (e.g. initiates admin client), split the parts to multiple test methods.

But ultimately in unit tests, they should be mocked if those slow part exists.